src/core/monte_carlo_scan_matcher.h

```diff
@@ -10,15 +20,37 @@
 #include "grid_scan_matcher.h"
 
 // TODO: move publish transform to observer
+
+/**
+ * \brief Scan Matcher that uses Mone Carlo Method
+ * The focus of scan matcher is to compare a scan and a built map. In this
+ * class the method of comparsion is based on Mone-Carlo method
+ */
 class MonteCarloScanMatcher : public GridScanMatcher {
 public:
   using ObsPtr = std::shared_ptr<GridScanMatcherObserver>;
 public:
+
+  /**
+   * \brief Initializes the scan matcher with a certan scan cost estimator
+   * \param estimator Current estimator of Scan Cost
+   * \param failed_iter Maximum amount of failed tries allowed
```

Comment:
We need a detailed description of both failed_iter. What doest the "failed try" mean?